### DISCUSSION TITLE
Introduce a (vert short) delay in the confirmation dialog before enabling encryption in rooms

### DISCUSSION THREAD

**Discussion by augusto-herrmann** [element-hq/element-meta#1713] [Ideas]
Enabling encryption in rooms cannot be undone. Any sensible UI should make sure the user is well informed and ask confirmation before applying any action that cannot be undone.

Currently, clients do explain that the process is reversible in a modal dialog that prompts for confirmation. However, the button can be clicked immediately as the dialog appears. If someone accidentally double-clicks (or double taps on mobile), they could easily encrypt a room by accident.

## Use case

Many people use Element for integrating via a bot bridge to other chat-based communities (e.g. Slack, Discord, Telegram, etc.) and having access to those communities using a free and open source app (namely, Element) instead of the closed proprietary ones.

The problem is that bot bridges usually require the room to be (and remain) unencrypted. When any user on the Matrix room can just permanently enable encryption by just clicking a button, sometimes by accident, without requiring any sort of confirmation, it will just break the bridge bot integration. Therefore making the room useless, requiring extra work for creating a new room and rebuilding the bridge bot integration all over again, not to mention that everyone accessing that external community from Matrix has to leave the now useless encrypted room and rejoin the new unencrypted room.

## Proposal

Introduce a very small delay (**one second** would be enough) between the moment when the modal dialog is displayed and the moment the confirmation button becomes available.

### Rationale

A delay of one second should not bother users who are sure of what they want to do, as they usually will take at least one second to activate the confirm anyway.

On the other hand, a delay of even one second will prevent unwanted activation when someone double taps the screen by accident – some may not have excellent motor skills and in mobile the confirmation button usually appears on the exact same position on the screen as the "encrypt room" button. It also does not help that the "encrypt room" button is the first option on the room properties screen and appears right in the middle. Not a good UI positioning choice for something irreversible.

**Note:** Edited in 5 June 2023 to reformulate the proposal.

**Comment by t3chguy** [comment]
<img width="986" alt="image" src="https://github.com/vector-im/element-meta/assets/2403652/b600b765-20bd-466d-b679-0d47202b2464">

Which client doesn't do this? I suggest opening an issue on their tracker @augusto-herrmann

**Reply by augusto-herrmann** [reply]
I had just enabled encryption by accident on a room using Element for Android 1.5.32. I had had the impression that it didn't ask for confirmation.

However, I have just created a room for the specific purpose of testing this, and found out that it does indeed ask for confirmation. I guess I must have accidentally double tapped on the middle of the screen. Sorry, that was my fault.

So I would change my UI suggestion to this instead: have a delay of **one second** in the UI before the user can click "ok" on the confirmation prompt, before enabling encryption. That would be useful for all clients in order to avoid accidents. Should I open an issue on each one of the clients repositories?

**Reply by t3chguy** [reply]
If you want a forced delay then such a discussion can be open here, I'd suggest re-phrasing the title

**Reply by augusto-herrmann** [reply]
Thanks! I've just edited and reformulated the proposal.
